Merge meteordemod into master

Cadair opened the pull request:
This Pull Request merges all the work on meteordemod etc to master. It's open as a way to review it all in one place. 

Todo list before merge:
- [x] Migrate the install scripts into ansible. This is primarily to ensure that people can upgrade safely and that we can be confident that multiple runs of the install script work as expected.
- [x] Documentation and schemas for new settings.
- [x] Compilation settings and testing on various platforms.
- [x] More testing.

colinluthier committed:
Create noaa_histogram_data

colinluthier committed:
Update receive_noaa.sh

colinluthier committed:
Update noaa_histogram_data

colinluthier committed:
Update receive_noaa.sh

colinluthier committed:
Update receive_noaa.sh

colinluthier committed:
Update Capture.php

colinluthier committed:
Update noaa_histogram_data

colinluthier committed:
Update receive_noaa.sh

jekhokie committed:
Fix Some Bits/Pieces

Change filename (include extension), some formatting changes, and add
the channel name to the histogram title.

Mahallon committed:
Merge branch 'master' of https://github.com/jekhokie/raspberry-noaa-v2 into Histogram_enhancement_2

colinluthier committed:
Update histogram.sh

colinluthier committed:
Update histogram.sh

Mahallon committed:
Merge branch 'Histogram_enhancement_2' of https://github.com/colinluthier/raspberry-noaa-v2 into Histogram_enhancement_2

Mahallon committed:
Merge branch 'master' of https://github.com/jekhokie/raspberry-noaa-v2 into Histogram_enhancement_2

Mahallon committed:
Merge branch 'master' of https://github.com/jekhokie/raspberry-noaa-v2 into Histogram_enhancement_2

dom-robinson committed:
adding gnuradio wav capture workflow to noaa

dom-robinson committed:
moving hardcoded GAIN setting back to passed varaiable

dom-robinson committed:
Updating hard coded Gain to passed variable

Note that at the moment the workflow seems to want a much higher gain setting than rtl_fm. I think this will improve as we sort the low pass and add a band pass filter, but it works fine if you have a high gain for now - I use 48.0 which i set in the settings.yml

dom-robinson committed:
Typo - line 82 adding space

dom-robinson committed:
adding M2 GRC and cleaning up for good order

Mahallon committed:
Merge branch 'master' of https://github.com/jekhokie/raspberry-noaa-v2 into Histogram_enhancement_2

dom-robinson committed:
Adding NOAA-APT, and tidying a few elements of the gnuradio push

dom-robinson committed:
Merge branch 'gnuradio-noaa' of https://github.com/dom-robinson/raspberry-noaa-v2 into gnuradio-noaa

New edits to push

colinluthier committed:
Delete settings.yml.sample

colinluthier committed:
Create settings.yml.sample

colinluthier committed:
Merge pull request #34 from dom-robinson/gnuradio-noaa

Gnuradio noaa

colinluthier committed:
Add files via upload

colinluthier committed:
Merge pull request #36 from colinluthier/running_code

Running code

colinluthier committed:
Merge pull request #37 from colinluthier/colinluthier-Noaa-Merge

Colinluthier noaa merge

Commit:
Create meteor_demodulate_qpsk.sh

The program used alias for a program which is not installed on Raspberry. Probably used to be on Raspberry as part of the script but got removed as newer programs were integrated.

Commit:
Added alias for MeteorDemod

I've added a new alias for MeteorDemod program created by Digitelektro: https://github.com/Digitelektro/MeteorDemod
I've also removed alias for meteor_demod since that program is no longer used.

Commit:
Added MeteorDemod to RTL_FM section

I've added MeteorDemod to RTL_FM section. TO BE TESTED THOROUGLY!

Commit:
Changed from GNU Radio to rtl_fm for Meteor M2

I've changed Meteor M2 reception method to be rtl_fm. That's until I integrate MeteorDemod to GNU Radio. I've only had it succesfully integrated it to rtl_fm section.

Commit:
Changed image names for Meteor

I've changed image names for Meteor. As a result, I've had to change which image names PHP has to look for in a database to display.

Commit:
Update Pass.php

Commit:
Changed image names for Meteor and NOAA

I've changed image names for Meteor and NOAA so that if it's daylight pass, the page displays image in visible light. At night, it displays IR image.

Commit:
Changed image names for Meteor and NOAA

I've changed image names for Meteor and NOAA so that if it's daylight pass, the page displays image in visible light. At night, it displays IR image.

Commit:
Added installation for MeteorDemod

I've added a way to install MeteorDemod. TO BE TESTED THOROUGLY!

Commit:
Added MeteorDemod .deb binary package

I've added a .deb binary for Meteor Demod provided by DigitElektro at: https://github.com/Digitelektro/MeteorDemod/releases/tag/2.0.0
I've also added a custom setting.ini file for MeteorDemod which works well.

Commit:
Change latitude and longtitude for MeteorDemod settings.ini file

I've added a latitude and longtitude variable for ansible to change in settings.ini file. I'm not familiar with Ansible so someone please correct my change. This file needs to be copied to `~/.config/meteordemod` so that folder has to have a file called settings.ini
I've added simple move command in `install_and_upgrade.sh` script to move that file to `~/.config/meteordemod`, but someone would have to manually input the correct coordinates every time someone runs `install_and_upgrade.sh` script since that would overwrite the file each time.

Commit:
Rename settings.ini to settings.ini.j2

Commit:
Copy MeteorDemod settings.ini file

I've added what I think WOULD work using Ansible to copy MeteorDemod settings.ini file into `~/.config/meteordemod` folder.

Again I'm not familiar with Ansible so please review this change and test it THOROUGLY!

Commit:
Update install_and_upgrade.sh

Commit:
Delete settings.ini

Commit:
Added meteor_demod alias

I've corrected file path for meteor_demod program. It is in `/usr/local/bin`, not in `/usr/bin/` as previously stated.

Commit:
Added underscore back to $METEOR_DEMOD

I made a mistake of removing an underscore. $METEORDEMOD is MeteorDemod by Digitelektro, here we want to use $METEOR_DEMOD which is meteor_demod, a completely different program.

dom-robinson committed:
Merge pull request #492 from mihajlo2003petkovic/master

Integrating MeteorDemod into rtl_fm part of receive_meteor.sh

Commit:
Removing MeteorDemod installation from deb package

I've removed installation from deb package since it is better to compile it from source so that MeteorDemod installs properly in case a user other than the one named "pi" tries to install it on the system.

Commit:
Delete Meteordemod-2.0.0-Linux.deb

Removing deb package for MeteorDemod since we'll be installing it directly from the source.

Commit:
Uploading MeteorDemod program part 1

Commit:
Uploading MeteorDemod program part 2

Commit:
Uploading MeteorDemod program part 3

Commit:
Creating a script to install MeteorDemod

Commit:
Adding a step to compile and install MeteorDemod from source

Commit:
Added projection variable for Meteor M2

I've added three variables so that the user can select which images they wish MeteorDemod to create.

Commit:
Projections selector for Meteor M2 images

I've changed variables so that Ansible can check from config.yml file which projections to create for Meteor M2.

Commit:
Added additional support for Mercator projections

If a user wishes to change `config.yml` file and select MeteorDemod to also create Mercator projection, now the webpage will also display that. I've essentially just added four more images for a daylight pass (truecolour, falsecolour, IR and heat map), and two more for a night pass (IR and heat map).

Commit:
Changed permissions in Ansible from Pi to root

I've changed permissions in Ansible from 0644 to 0755 which would hopefully fix the issue of it not being able to write to `~/.config`/meteordemod directory.

Commit:
Correct owner from pi to root

Commit:
Send every image to Discord and email --- TO BE TESTED

I updated my `receive_meteor.sh` script so it will send every Meteor image it receives to Discord and email using a simple for loop. This has to be tested.

Commit:
Removed unnecessary comments used when testing

Commit:
Deleted `meteor_decode_bitstream.sh` since it is not used anymore

Commit:
Changed permissions

Commit:
Changed owner from root to $USER of config files

I changed the owner from root to $USER for config files for MeteorDemod so Ansible can have permissions to copy variables to that location.

Commit:
Setting `~/.config/meteordemod` ownership to the current user

I've set `~/.config/meteordemod` ownership to the current user so that Ansible can update that folder with `settings.ini` file.

Commit:
Changed order of how the images will be displayed on the webpage

Commit:
Flip spreaded images at night passes

dom-robinson committed:
Merge pull request #493 from mihajlo2003petkovic/master

Compiling MeteorDemod from source + added support for Mercator projection on webpanel

Commit:
Fix indentations in rtl_fm section

Commit:
Fixed Discord pushing

Commit:
Fixing Discord pushing part 2

Commit:
Removed the unnecessary log added for testing

Commit:
Moved the whole Discord and email pushing section to the end of the script

Commit:
Fixed GNU Radio section. It finally works with MeteorDemod!!!

I've just fixed GNU Radio section. Now MeteorDemod works on both GNU Radio and RTL_FM!

Commit:
Dried up by reusing code in `receive_meteor.sh` A LOT

I've dried up the code A LOT in `receive_meteor.sh` script since I've reused common sections like annotating and moving images to `/srv/images` and inserting necessary data into a database.

dom-robinson committed:
Merge pull request #497 from mihajlo2003petkovic/master

Succesfully integrated MeteorDemod into GNU Radio section

Commit:
Fixed Discord and email pushing so that every image gets pushed

Commit:
Delete noaa_bd.sh

Commit:
Delete noaa_cc.sh

Commit:
Delete noaa_he.sh

Commit:
Delete noaa_hf.sh

Commit:
Delete noaa_hvc.sh

Commit:
Delete noaa_hvc_precip.sh

Commit:
Delete noaa_hvct.sh

Commit:
Delete noaa_hvct_precip.sh

Commit:
Delete noaa_jf.sh

Commit:
Delete noaa_jj.sh

Commit:
Delete noaa_lc.sh

Commit:
Delete noaa_mb.sh

Commit:
Delete noaa_mcir.sh

Commit:
Delete noaa_mcir_precip.sh

Commit:
Delete noaa_md.sh

Commit:
Delete noaa_msa.sh

Commit:
Delete noaa_msa_precip.sh

Commit:
Delete noaa_no.sh

Commit:
Delete noaa_sea.sh

Commit:
Delete noaa_ta.sh

Commit:
Delete noaa_therm.sh

Commit:
Delete noaa_wv.sh

Commit:
Delete noaa_za.sh

Commit:
Added a single file for every NOAA enhancement

Every enhancement creation is now handled in a single file.

Commit:
Decreased code size

Now every enhancement creation is handled in a single `noaa_enhancements.sh` file.

dom-robinson committed:
Merge pull request #500 from mihajlo2003petkovic/master

Fixed Discord and email pushing so that every image gets pushed + simple indentation fix

dom-robinson committed:
tidying up install on buster

dom-robinson committed:
new noaa enhancements script needed chmod +x

dom-robinson committed:
specifying the python2 interpreter to be used

dom-robinson committed:
Adding annotation and sample changes requested by Vince

dom-robinson committed:
cosmetic cleanup for enhancments in annotations

dom-robinson committed:
minor tweaks to display antenna details in annotation

dom-robinson committed:
reverting the annotation changes

dom-robinson committed:
ammending python for M2 capture

dom-robinson committed:
permission for noaa script and histogram

dom-robinson committed:
permissions on script

dom-robinson committed:
Adding note about apt update

Note that to install cmake under bullseye you NEED to run sudo apt-get update BEFORE you try to install.

dom-robinson committed:
forcing apt update to ensure cmake installs

dom-robinson committed:
changing shebang to sort gnuradio and python version in bullseye

dom-robinson committed:
commit error - correcting

dom-robinson committed:
buster histogram output naming

Cadair commented on install_MeteorDemod.sh:
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
My first major comment here is that this script shouldn't exist it should be done by ansible. Which I realise is the same as me offering to make that happen :wink:

Cadair replied:
To reiterate, following discussions on Discord, I *really* think we need to make this swap before we merge this.

Cadair commented on ansible/roles/webserver/tasks/main.yml:
```diff
@@ -14,6 +14,26 @@
+
+- name: install the package, force upgrade
+  pip: 
+    name: numpy
```
what in the webserver needs numpy?

dom-robinson replied:
the polar pass projections..

Cadair replied:
Where does php shell out to python?!

Cadair commented on ansible/roles/webserver/tasks/main.yml:
```diff
@@ -14,6 +14,26 @@
     state: present
   when: raspbian_version.stdout == 'stretch'
 
+- name: php repository gpg key (buster only)
+  become: yes
+  apt_key:
+    url: "https://packages.sury.org/php/apt.gpg"
+    state: present
+  when: raspbian_version.stdout == 'buster'
+
+- name: php repository (buster only)
+  become: yes
+  apt_repository:
+    repo: deb https://packages.sury.org/php/ buster main
+    filename: php7.list
+    state: present
+  when: raspbian_version.stdout == 'buster'
```
So I understand, this is an alternative fix to #491 which uses a third party php repo instead of installing a different version of php?

dom-robinson replied:
yep.

Cadair commented on install_and_upgrade.sh:
```diff
@@ -85,6 +85,9 @@ if [ $? -ne 0 ]; then
   fi
 fi
 
+log_running "Setting ~/.config/meteordemod ownership to the current user"
+sudo chown $USER:$USER -R ~/.config/meteordemod
```
again, ansible should be doing this.

Cadair commented on software/MeteorDemod/CMakeLists.txt:
```diff
@@ -0,0 +1,135 @@
+cmake_minimum_required(VERSION 3.5)
```
I am also *seriously* not ok with just including the whole source code for MeteorDemod. Why can't we just clone this from upstream?

dom-robinson replied:
no strong opinion from my side - if it works im happy, but if there is a better way then also happy!

Cadair replied:
It looks like all of @mihajlo2003petkovic 's changes to upstream have been merged so I see no reason why we can't clone upstream.

Cadair commented on ansible/roles/common/tasks/dependencies.yml:
```diff
@@ -15,6 +15,7 @@
       - imagemagick
       - jq
       - libatlas-base-dev
+      - libgfortran5
```
You have added this here but not deleted the stretch only block below like I did in #491

dom-robinson replied:
agree - i did that manually and forgot to remove stretch

Cadair commented on ansible/roles/common/templates/settings.ini.j2:
```diff
@@ -0,0 +1,64 @@
+[Program]
```
This file should be renamed to be `meteordemod_settings.ini.j2` and the ansible bit updated to use that new local filename (without changing the "remote" one where it gets installed to).

Cadair commented on scripts/image_processors/noaa_enhancements.sh:
```diff
@@ -9,6 +9,7 @@
 #   1. Map overlap file
```
Well about time someone purged all these files lol

Cadair commented on scripts/receive_meteor.sh:
```diff
@@ -112,51 +112,20 @@ if [ "$METEOR_RECEIVER" == "rtl_fm" ]; then
   log "Starting rtl_fm record" "INFO"
   ${AUDIO_PROC_DIR}/meteor_record_rtl_fm.sh $CAPTURE_TIME "${RAMFS_AUDIO_BASE}.wav" >> $NOAA_LOG 2>&1
 
+  sleep 2
```
Why? Can we add a comment here explaining these otherwise some smartass like me will remove them.

(Also can't we test for something rather than just sleep? i.e. "if file doesn't exist sleep")

dom-robinson replied:
either way. happy to see that improved - i think it just takes a second for the file to close..

Cadair reviewed (commented):
I shall install this on my pi and have a play, but initial comments here.

Thanks everyone for your work on this, it is a great feature and contains some very pleasing code tidying.

dom-robinson committed:
chmods no longer needed for install_Meteordemod.sh

dom-robinson committed:
Credits 

@Digitalelectro and @mihajlo2003petkovic will also get release-note mention in due course!

Cadair commented:
We aren't using the old meteor_demod any more correct? If that's true we should remove all the references to it, especially where it's installed here: https://github.com/jekhokie/raspberry-noaa-v2/blob/master/ansible/roles/common/tasks/tools.yml#L9

MihajloPi commented:
> We aren't using the old meteor_demod any more correct? If that's true we should remove all the references to it, especially where it's installed here: https://github.com/jekhokie/raspberry-noaa-v2/blob/master/ansible/roles/common/tasks/tools.yml#L9

No, we are using BOTH meteor_demod and meteordemod. What we're not using anymore is medet_arm. meteor_demod is used in RTL_FM section to create s files from wav recordings. We could directly feed meteordemod wav files, but I found out that we get better images by first converting wav to s files using meteor_demod, and feeding them to meteordemod. Also, meteor_demod converts wav to s files MUCH FASTER than meteordemod, which significantly speeds up the whole decoding process.

Cadair commented:
Right ok, so we should pull the medet_arm install stuff as we aren't using it anymore?

(Also more reason to purge rtl_fm eventually :wink: )

MihajloPi commented:
> Right ok, so we should pull the medet_arm install stuff as we aren't using it anymore?
> 
> (Also more reason to purge rtl_fm eventually :wink: )

I don't think that we should purge RTL_FM at all since many users are still using it, including myself. It is also good for developing since it always worked as I've had problems with GNU Radio on Bullseye etc. We should always leave it as a backwards compatibility.

Cadair commented:
This isn't the place to discuss that :wink:

ve3elb commented:
@Mihajlo I fully agree with you. RTL_FM should never be removed as no everyone likes using GNUradio. Both should stay as a choice for users to use. 



> On Jan 24, 2022, at 11:20 AM, Mihajlo Petković ***@***.***> wrote:
> 
> ﻿
> Right ok, so we should pull the medet_arm install stuff as we aren't using it anymore?
> 
> (Also more reason to purge rtl_fm eventually 😉 )
> 
> I don't think that we should purge RTL_FM at all since many users are still using it, including myself. It is also good for developing since it always worked as I've had problems with GNU Radio on Bullseye etc. We should always leave it as a backwards compatibility.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.

Cadair commented:
Seriously, can we please have this discussion somewhere else...

dom-robinson committed:
another attempt to sort buster histograms

dom-robinson committed:
Merge branch 'meteordemod-staging' of https://github.com/jekhokie/raspberry-noaa-v2 into meteordemod-staging

dom-robinson committed:
reorder display of m2

dom-robinson committed:
Merge branch 'meteordemod-staging' of https://github.com/jekhokie/raspberry-noaa-v2 into meteordemod-staging

dom-robinson committed:
Fixing failover for recording priority for gnuradio captures

colinluthier commented:
I started using this today on my Pi Zero 2. I have not had a capture yet but I have noticed a few things. 

```
meteor_create_spreaded_images_without_overlay
meteor_create_equidistant_projection
meteor_create_mercator_projection
```
are not tested before running Ansible.

Install_MeterDemod.sh will fail on the Zero 2. Changing the flag to make -j2 fixes it. Increasing the swap file size might work too.

Commit:
Added rain overlay variable

Commit:
Added rain overlay variable for MeteorDemod

Commit:
Cloning and installing MeteorDemod directly from the Github

Commit:
Removing MeteorDemod directory since we are cloning it directly from the Github

Commit:
Updated Capture.php so that it accepts rain overlay created by MeteorDemod

dom-robinson committed:
Merge pull request #513 from mihajlo2003petkovic/meteordemod-staging

Cloning MeteorDemod directly from the Github + added rain overlay variable

colinluthier committed:
Merge branch 'master' into meteordemod-staging

Cornmeister commented:
at the settings.yml it looks like some new settings havent got a description/comment yet

dom-robinson committed:
Celestrack updates

Celestrack URLs have been updated - so have wget commands.

dom-robinson committed:
single quote prune scripts

MihajloPi commented:
Fixed yesterday in beta-development as this was still based for Meteor M2 reception which is dead, and merged from beta-development into master.
